pyam/core.py

```diff
@@ -1284,6 +1287,48 @@ def export_meta(self, excel_writer, sheet_name='meta'):
         if close:
             excel_writer.close()
 
+    def to_datapackage(self, path, tmp='tmp'):
+        """Write object to a frictionless datapackage
+
+        More information: https://frictionlessdata.io
+
+        Returns the saved :class:`datapackage.Package`. When adding metadata
+        (descriptors), please follow the `template` defined by
+        https://github.com/OpenEnergyPlatform/metadata
+
+        Parameters
+        ----------
+        path: string
+            file path
+        tmp: string, default 'tmp'
+            name for temporary folder to export csv files
+            (deleted after saving datapackage)
+        """
+
+        path, tmp = Path(path), Path(tmp)
+        if tmp in list(tmp.parent.iterdir()):
+            msg = 'folder `{}` already exists in this directory!'
+            raise ValueError(msg.format(tmp))
+
+        # save data and meta tables to a temporary folder
+        tmp.mkdir()
+        self.data.to_csv(tmp / 'data.csv', index=False)
+        self.meta.to_csv(tmp / 'meta.csv')
+
+        # cast tables to datapackage
+        package = Package()
+        package.infer('{}/*.csv'.format(tmp))
```

Comment:
Maybe you have it elsewhere, but the documentation for the metadata is not covered here. I guess there is some magic with the `meta.csv` file? Hard to know what to do to change e.g. the license for newcomers.

Reply:
The `meta` table is used for categorization and quantitative indicators of the scenarios. For the IPCC SR15, we used that for warming categories (Below 1.5°C, 1.5°C with low overshoot, etc.) or cumulative CO2 emissions over the century. [Read the docs page](https://pyam-iamc.readthedocs.io/en/stable/data.html#the-meta-table)

Currently, `pyam` doesn't support metadata in the frictionless sense (license, sources, etc.).  That's why I implemented `to_datapackage()` such that it returns the `Package` (with only data and meta table), and it's up to the user to add descriptors in his/her workflow.

If this is picked up by a broader range of users with more advanced requirements like proper metadata handling, I'd be happy to discuss implementation. But as a pragmatic first step, I'd not add it to this PR.